py/README.md

```diff
@@ -0,0 +1,374 @@
+# AllOfUs Python Client Library
+
+Python code in this library is used to access the AllOfUs Workbench API from
+the context of a Jupyter notebook. 
+
+## Installing the library
+
+The latest version of the library will be installed automatically on the Jupyter
+server your notebook code runs in, but if you need to upgrade to a version which 
+has not been released yet, you can do so by running in a notebook cell:
+
+```
+!pip install --user --upgrade 'https://github.com/all-of-us/pyclient/archive/<VERSION>.zip#egg=aou_workbench_client&subdirectory=py'
+```
+
+or for Python 2.7:
+
+```
+!pip2.7 install --user --upgrade 'https://github.com/all-of-us/pyclient/archive/<VERSION>.zip#egg=aou_workbench_client&subdirectory=py'
+```
+
+and then restarting your Python kernel.
+
+The modules you will use in your code are in the aou_workbench_client package.
+
+## Using cohorts in a notebook
+
+The aou_workbench_client.cohorts module provides functions for materializing cohorts.
+
+### materialize_cohort
+
+materialize_cohort is used to fetch some or all results of cohort materialization, 
+and return a generator of dictionaries representing the results.
+
+This can be used to retrieve information about some or all of the participants in a
+cohort you defined in the AllOfUs workbench.
+
+No server call takes place to retrieve the data until you start iterating over
+the generator.
+
+For not-too-large result sets, it is reasonable to use a Pandas `DataFrame` 
+or call to `list()` to bring all the results from the generator into memory at once. 
+If the result sets are very large, you may need to stream the results to disk as you 
+use them to avoid running into out of memory errors.
```

Comment:
Are we really in danger of memory issues? If so, dask is probably the way to go. It would allow users to keep using DataFrames, but have them work out-of-core.